src/scanpy/tools/_umap.py

```diff
@@ -146,6 +147,8 @@ def umap(
         :attr:`~anndata.AnnData.obsp`\\ ``[.uns[neighbors_key]['connectivities_key']]`` for connectivities.
     copy
         Return a copy instead of writing to adata.
+    parallel
```

Comment:
On the other hand:

```python
sc.tl.umap(adata, parallel=True, random_state=42)
```

works so I think this needs a test + updated comment to reflect whatever is supposed to be going on here.

Reply:
As does

```python
sc.tl.umap(adata, parallel=True, random_state=np.random.RandomState(42))
```

Reply:
I added a test to see if it errors (it doesn't)

Reply:
Yes, but we should also warn users about this random state business.  And check that the warning is raised every time `parallel` is True because this is very important!

Reply:
I also thing the docs in umap is wrong. The speedup stays if you submit a randomseed

Reply:
Ok! But then we should _definitely_ warn in that case.

Reply:
I added a warning

Reply:
ok there is in issue with `check_random_state` as it transforms `None` into a seed.

Reply:
ok the function is bugged on the umap side. If you force `None` as the random seed, it errors. I'll make an issue with umap for clarification. I'll but the PR on ice while i wait for an update.

Reply:
https://github.com/lmcinnes/umap/issues/1155